spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala

```diff
@@ -58,6 +58,53 @@ class CometExecSuite extends CometTestBase {
     }
   }
 
+  // TODO: Add a test for SortMergeJoin with join filter after new DataFusion release
+  test("SortMergeJoin without join filter") {
+    withSQLConf(
+      SQLConf.ADAPTIVE_AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
+      withParquetTable((0 until 10).map(i => (i, i % 5)), "tbl_a") {
+        withParquetTable((0 until 10).map(i => (i % 10, i + 2)), "tbl_b") {
+          val df1 = sql("SELECT * FROM tbl_a JOIN tbl_b ON tbl_a._2 = tbl_b._1")
+          checkSparkAnswerAndOperator(df1)
```

Comment:
Should we add checks to make sure the plan includes the Comet SMJ, i.e. `stripAQEPlan(df.queryExecution.executedPlan).collectFirst...`

Reply:
`checkSparkAnswerAndOperator` already does the check. If there is Spark SMJ or other join operators, it will report error.

Reply:
I see. How does `checkSparkAnswerAndOperator` checks whether it is Comet SMJ? 
Unless we provide `includeClasses`, it only checks classes below?
https://github.com/apache/arrow-datafusion-comet/blob/main/spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala#L156
```
      case _: CometScanExec | _: CometBatchScanExec => true
      case _: CometSinkPlaceHolder | _: CometScanWrapper => false
      case _: CometExec | _: CometShuffleExchangeExec => true
      case _: CometBroadcastExchangeExec => true
      case _: WholeStageCodegenExec | _: ColumnarToRowExec | _: InputAdapter => true
```

Reply:
`CometSortMergeJoinExec` along with other native operators are `CometExec` so it is listed in `checkCometOperators`. We don't white list all native operators but the common base class.

Reply:
Got it, thanks. I think it is good for now. Ideal if we can check `CometSortMergeJoinExec` specifically, as we will add other joins. In that way, we can make sure that we are not testing different join accidentally.

Reply:
Ah, that makes sense. I can add something to verify join type (sort merge join or hash join or others) in a follow up. Thanks for the suggestion.